.vscode/settings.json

```diff
@@ -1,6 +1,11 @@
 // Place your settings in this file to overwrite default and user settings.
 {
-  "python.formatting.provider": "black",
+  "[python]": {
+    "editor.defaultFormatter": "ms-python.black-formatter"
+  },
+  "[scm]": {
+    "editor.defaultFormatter": "AndreasArvidsson.andreas-talon"
```

Comment:
I wonder if we want to require all Cursorless developers to use the `andreas-talon` extension.  Probably not a big deal, and we can probably revisit fairly easily, but feels a bit opinionated.  Eg your talon formatter, tho that's off by default, right?

I wonder if the scm formatter should be its own extension. We could also add cool stuff like function signature hints for our query predicates, hover / go to definition for those, etc.  Tho given they're custom to us, I wonder if separate extension makes sense or if it should really be part of cursorless developer tools

Another advantage of having your scm formatter be separate extension is you can give it a pre-commit config so we can easily use it in CI

But I'm not opposed to merging as is just to get the ball rolling and revisit later

Reply:
I definitely feel that this is a topic we should revisit. I like many of those suggestions.

Reply:
So when people install your extension it's going to reformat the talon files? That's not great, because we use wen's formatter in pre-commit

Reply:
Only if they run the formatter. Format on save is not default in vscode. The readme describes how to deactivate format on save for Talon files specifically. We could put that setting in the repository vscode file?

Reply:
Yeah I'm pretty sure I had to manually deactivate the formatter when I first installed your extension

Maybe we could just have a config that sets wen's extension as default formatter for talon files like we're having yours for scm?

Also, maybe we should make the scm syntax colouring extension a recommended extension while we're here?

Reply:
We can definitely do that setting in Cursorless.
Maybe.

Reply:
Wens formatter is now the default for Talon files.

Reply:
I mean do we want to add `jrieken.vscode-tree-sitter-query` as a recommended extension

Reply:
I guess we could. Do we have other extension recommendations? if yes then where are they? and if know why this one in particular?

Reply:
https://github.com/cursorless-dev/cursorless/blob/main/.vscode/extensions.json

looks like we recommend our autoformatters; maybe we should add the new autoformatters as well?

Reply:
Sounds reasonable. What exactly are we missing?

Reply:
- `AndreasArvidsson.andreas-talon`
- `wenkokke.tree-sitter-talon`
- `jrieken.vscode-tree-sitter-query`

Reply:
done